master/security/icmp-ping.md

```diff
@@ -0,0 +1,136 @@
+
+- Allow ICMP ping, but only for workloads, host endpoints (or both)
+- Allow ICMP for pods launched by operators for diagnostic purposes, but block other uses
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMPv4 and/or ICMPv6
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with:
+
+- Protocol match for ICMPv4 and ICMPv6
+- icmp/NotICMP match for ICMP type and code
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow ICMP traffic based on specific types and codes. For example, you can specify ICMP type 5, code 2 to match specific ICMP redirect packets. 
+
+For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+- [Deny all ICMP, all workloads](#deny-all-icmp, all workloads)
+- [Allow ICMP ping, all workloads and host endpoints](#allow-icmp-ping-all-workloads-and-host-endpoints)
+- [Allow ICMP matching protocol type and code, all workloads](#allow-icmp-matching-protocol-type-and-code-all-workloads)
+
+#### Deny all ICMP, all workloads
+
+In this example, we introduce a "deny all ICMP" **GlobalNetworkPolicy**. 
+
+This policy **selects all workloads that match a selector**. It enables a default deny for all workloads, in addition to the explicit ICMP deny rules specified in the policy. For more on default deny policy, see [Calico Network Policy]({{site.baseurl}}/{{page.version}}/security/caliconetworkpolicy).   
+
+If your ultimate goal is to allow some traffic, have your regular "allow" policies in place before applying a global deny-all ICMP traffic policy.
+
+In this example, all workloads are blocked from sending or receiving **ICMPv4** and **ICMPv6** messages. Because a GlobalNetworkPolicy applies to hostendpoints as well, to limit the policy to only workloads, we use the selector **projectcalico.org/orchestrator == 'kubernetes'**.
+
+If **ICMPv6** messages are not used in your deployment, it is still good practice to deny them specifically as shown below. 
+
+In any "deny-all" Calico network policy, be sure to specify a lower order (**order:200**) than regular policies that might allow traffic.   
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: block-icmp
+spec:
+  order: 200
+  selector: projectcalico.org/orchestrator == 'kubernetes'
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Deny
+    protocol: ICMP
+  - action: Deny
+    protocol: ICMPv6
+  egress:
+  - action: Deny
+    protocol: ICMP
+  - action: Deny
+    protocol: ICMPv6    
+```
+
+#### Allow ICMP ping, all workloads and host endpoints
+
+In this example, workloads can receive **ICMPv4 type 8** and **ICMPv6 type 128** ping requests.
```

Comment:
> **ICMPv4 type 8** and **ICMPv6 type 128** ping requests.

"ping requests originating from other workloads and host endpoints".


(Since we have the `source -> selector: all()` specified, it limits the allowed traffic only from workloads and host endpoints)

Reply:
Fixed.